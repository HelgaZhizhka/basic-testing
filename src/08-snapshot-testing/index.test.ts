import { generateLinkedList } from './index';

interface TechNode {
  value: string | null;
  next: TechNode | null;
}

describe('generateLinkedList', () => {
  let techList: string[];
  let expectedTechNode: TechNode;

  beforeEach(() => {
    techList = [
      'JavaScript',
      'React',
      'Vue',
      'Node.js',
      'Webpack',
      'Jest',
      'TypeScript',
      'CSS',
      'HTML',
      'GraphQL',
    ];

   expectedTechNode = {
     value: 'JavaScript',
     next: {
       value: 'React',
       next: {
         value: 'Vue',
         next: {
           value: 'Node.js',
           next: {
             value: 'Webpack',
             next: {
               value: 'Jest',
               next: {
                 value: 'TypeScript',
                 next: {
                   value: 'CSS',
                   next: {
                     value: 'HTML',
                     next: {
                       value: 'GraphQL',
                       next: {
                         value: null,
                         next: null,
                       },
                     },
                   },
                 },
               },
             },
           },
         },
       },
     },
   };


  });

  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const resultTechNode = generateLinkedList(techList);
    expect(resultTechNode).toStrictEqual(expectedTechNode);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const resultTechNode = generateLinkedList(techList);
    expect(resultTechNode).toMatchSnapshot();
  });
});
