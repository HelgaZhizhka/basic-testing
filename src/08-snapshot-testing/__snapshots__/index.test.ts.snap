// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": {
            "next": {
              "next": {
                "next": {
                  "next": {
                    "next": {
                      "next": null,
                      "value": null,
                    },
                    "value": "GraphQL",
                  },
                  "value": "HTML",
                },
                "value": "CSS",
              },
              "value": "TypeScript",
            },
            "value": "Jest",
          },
          "value": "Webpack",
        },
        "value": "Node.js",
      },
      "value": "Vue",
    },
    "value": "React",
  },
  "value": "JavaScript",
}
`;
